CMakeLists.txt

```diff
@@ -367,15 +367,16 @@ unset(_VAST_ENABLE_ASSERTIONS_DEFAULT)
 # -- sanitizers ----------------------------------------------------------------
 
 # Address Sanitizer
-if ("${CMAKE_BUILD_TYPE}" IN_LIST "Debug;CI")
+set(_vast_build_types_with_default_asan "Debug;CI")
+if (CMAKE_BUILD_TYPE IN_LIST _vast_build_types_with_default_asan)
   set(_VAST_ENABLE_ASAN_DEFAULT ON)
 else ()
   set(_VAST_ENABLE_ASAN_DEFAULT OFF)
 endif ()
 
 option(VAST_ENABLE_ASAN
        "Insert pointer and reference checks into the generated binaries"
-       _VAST_ENABLE_ASAN_DEFAULT)
+       "${_VAST_ENABLE_ASAN_DEFAULT}")
```

Comment:
This bug essentially makes us wait on tenzir/actor-framework#6 before merging this, otherwise the local development experience suffers on macOS.